tests/license_header_check.py

```diff
@@ -0,0 +1,105 @@
+# SPDX-License-Identifier: LGPL-2.1-or-later
+
+import re
+import sys
+from pathlib import Path
+
+paths = ["rust/src", "tests"]
+
+headers = [
+    "#\\s|\\SSPDX-License-Identifier: LGPL-2.1-or-later",
+    "//\\s|\\SSPDX-License-Identifier: Apache-2.0",
+]
+
+excludes = [
+    "**/*.json",
+    "**/*.txt",
+    "**/*.in",
+    ".gitignore",
+    ".gitmodules",
+    "LICENSE",
+    "NOTICE",
+    "go.mod",
+    "go.sum",
+    ".git/*",
+    "logo/*",
+    "licenses",
+    "**/*.md",
+    "**/*.lock",
+    "**/*.toml",
+    "**/*.yml",
+    "tests/integration/test_802.1x_srv_files/*",
+    "tests/integration/test_ipsec_certs/*",
+    "tests/integration/test_captures/*",
+    "rust/target/*",
+    "rust/src/python/dist/*",
+    "rust/src/python/build/*",
+    "rust/src/python/nmstate.egg-info/*",
+    "examples/*",
+    "packaging/*",
+]
+
+total_files = 0
+files_with_header = 0
+files_without_header = 0
+excluded_files = 0
+
+
+def get_included_files():
+    global total_files
+    global excluded_files
+    directory_paths = (Path(directory) for directory in paths)
+    files = []
+    for directory_path in directory_paths:
+        for file in directory_path.glob("**/*"):
+            if file.is_dir():
+                continue
+
+            total_files += 1
+
+            if any(file.match(pattern) for pattern in excludes):
+                excluded_files += 1
+                continue
+
+            files.append(file)
+
+    return files
+
+
+print(
+    "------------------------- License Header Check Start -------------------------"
+)
+patterns = [re.compile(header) for header in headers]
+
+for file_path in get_included_files():
+
+    with open(file_path, "r") as file:
+        content = file.readline()
+
+        # Read second line if the first is a shebang
+        if content.startswith("#!"):
+            content = file.readline()
+
+        matches = [re.search(pattern, content) for pattern in patterns]
```

Comment:
Here you should us `re.match`, not `re.search`, because we want to match it at the beginning of the line, not at any point of the line. Or better, use `re.fullmatch`.